lib/active_admin/router.rb

```diff
@@ -1,27 +1,21 @@
 module ActiveAdmin
+  # @private
   class Router
-    def initialize(application)
-      @application = application
+    attr_reader :namespaces, :router
+
+    def initialize(router:, namespaces:)
+      @router, @namespaces = router, namespaces
     end
 
-    # Creates all the necessary routes for the ActiveAdmin configurations
-    #
-    # Use this within the routes.rb file:
-    #
-    #   Application.routes.draw do |map|
-    #     ActiveAdmin.routes(self)
-    #   end
-    #
-    # @param router [ActionDispatch::Routing::Mapper]
-    def apply(router)
+    def apply(router = @router)
```

Comment:
@varyonic, not sure this commit  makes code more cleaner, I don't understand now if there   there are cases when different routers can be used for initialize and apply method ? in other words after router passed to constructor, why this method needs argument

Reply:
@varyonic  ignore this since it was fixed by next commit